### PR TITLE
fix: IaC errors display in description panel [ROAD-486]

### PR DESCRIPTION
**Pull request by ArtsiomCh**
image

### CODE REVIEW

File: src/integTest/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowPanelIntegTest.kt [hunk 1]
```diff
@@ -98,7 +113,38 @@ class SnykToolWindowPanelIntegTest : HeavyPlatformTestCase() {
...
+        assertEquals(iacError, toolWindowPanel.currentIacError)
+
+        TreeUtil.selectNode(toolWindowPanel.getTree(), toolWindowPanel.getRootIacIssuesTreeNode())
+        PlatformTestUtil.waitWhileBusy(toolWindowPanel.getTree())
+
+        val descriptionComponents = toolWindowPanel.getDescriptionPanel().components.toList()
+        assertTrue(descriptionComponents.any { it is SnykErrorPanel })
+
+        // tear Down
+        unmockkStatic("io.snyk.plugin.UtilsKt")
```
Comment (bastiandoetsch) [inline src/integTest/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowPanelIntegTest.kt:147]: This should be done in teardown, as it is never called if the assertion fail.

File: src/integTest/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowPanelIntegTest.kt [hunk 2]
```diff
@@ -98,7 +113,38 @@ class SnykToolWindowPanelIntegTest : HeavyPlatformTestCase() {
...
+        every { getIacService(project).isCliInstalled() } returns true
+        every { getIacService(project).scan() } returns iacResultWithError
+
+        // actual test run
+        project.service<SnykTaskQueueService>().scan()
+
+        val toolWindowPanel = project.service<SnykToolWindowPanel>()
+        PlatformTestUtil.waitWhileBusy(toolWindowPanel.getTree())
+
+        assertEquals(iacError, toolWindowPanel.currentIacError)
```
Comment (bastiandoetsch) [inline src/integTest/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowPanelIntegTest.kt:138]: Consider to use verify { ... } to make sure that a service call is actually happening and the description panel is populated by the result of the call, not from any other error.

File: src/integTest/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowPanelIntegTest.kt [hunk 3]
```diff
@@ -98,7 +113,38 @@ class SnykToolWindowPanelIntegTest : HeavyPlatformTestCase() {
...
+
+        TreeUtil.selectNode(toolWindowPanel.getTree(), toolWindowPanel.getRootIacIssuesTreeNode())
+        PlatformTestUtil.waitWhileBusy(toolWindowPanel.getTree())
+
+        val descriptionComponents = toolWindowPanel.getDescriptionPanel().components.toList()
+        assertTrue(descriptionComponents.any { it is SnykErrorPanel })
```
Comment (bastiandoetsch) [inline src/integTest/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowPanelIntegTest.kt:144]: Please check for the actual error content, to make sure it's not a different error displayed :). You can get the Label with UIComponentFinder class.

File: src/integTest/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowPanelIntegTest.kt (hunk 2 above)
Comment (ArtsiomCh) [reply src/integTest/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowPanelIntegTest.kt:138]: Here we check intermediate condition required for the test of the test to have sense.
Also I don't want to chain the test to any particulate service or method call. But instead would like to check that we have required final result: error shown in description panel.
PS But Thanks for clarification! will add check for the expected error text and not any random error.

File: src/integTest/kotlin/UIComponentFinder.kt [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import java.awt.Component
+import java.awt.Container
+import kotlin.reflect.KClass
+import kotlin.reflect.cast
+
+object UIComponentFinder {
+
+    @OptIn(ExperimentalStdlibApi::class)
```
Comment (bastiandoetsch) [inline src/integTest/kotlin/UIComponentFinder.kt:8]: Interesting, what's that?

File: src/integTest/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowPanelIntegTest.kt [hunk 5]
```diff
@@ -41,17 +43,23 @@ class SnykToolWindowPanelIntegTest : HeavyPlatformTestCase() {
         unmockkAll()
         resetSettings(project)
         setupDummyCliFile()
+        // restore modified Registry value
+        isIacEnabledRegistryValue.setValue(isIacEnabledDefaultValue)
+        unmockkStatic("io.snyk.plugin.UtilsKt")
```
Comment (bastiandoetsch) [inline src/integTest/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowPanelIntegTest.kt:48]: The unmockStatic looks redundant to unmockAll().

File: src/integTest/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowPanelIntegTest.kt [hunk 6]
```diff
@@ -41,17 +43,23 @@ class SnykToolWindowPanelIntegTest : HeavyPlatformTestCase() {
...
     }
 
     override fun tearDown() {
         unmockkAll()
         resetSettings(project)
         removeDummyCliFile()
+        // restore modified Registry value
+        isIacEnabledRegistryValue.setValue(isIacEnabledDefaultValue)
+        unmockkStatic("io.snyk.plugin.UtilsKt")
```
Comment (bastiandoetsch) [inline src/integTest/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowPanelIntegTest.kt:57]: Same here, why do you unmock utils explicitly and not just call unmockAll() at the end of the method?

File: src/integTest/kotlin/UIComponentFinder.kt (hunk 4 above)
Comment (ArtsiomCh) [reply src/integTest/kotlin/UIComponentFinder.kt:8]: It's easy to find, just comment it and Idea will show you ;)
Actually it's needed for KClass<T>.cast() call as it's for some reason still an experimental function

**Review by bastiandoetsch — APPROVED** [review]

**Pull request merged**